doc/reference/trd104-syscalls.md

```diff
@@ -596,12 +611,12 @@ The register arguments for Read-Write Allow system calls are as
 follows. The registers r0-r3 correspond to r0-r3 on CortexM and a0-a3
 on RISC-V.
 
-| Argument         | Register |
-|------------------|----------|
-| Driver number    | r0       |
-| Allow number     | r1       |
-| Address          | r2       |
-| Size             | r3       |
+| Argument         | Register | Type               | Value                                                            |
+|------------------|----------|--------------------|------------------------------------------------------------------|
+| Driver number    | r0       | `u32`              |                                                                  |
+| Allow number     | r1       | `u32`              |                                                                  |
+| Address          | r2       | `&mut [u32; {r3}]` | Pointer to array of userspace memory of length `{r3}` **or** `0` |
```

Comment:
I'm not sure we can/should be this specific for `address`. What would it mean to violate this? Unless the memory would go beyond the app's space then this would be satisfied.

Reply:
> I'm not sure we can/should be this specific for `address`. What would it mean to violate this? Unless the memory would go beyond the app's space then this would be satisfied.

I updated this to actually be slightly more specific, i.e. 
> Pointer to array of **writeable** userspace memory of length `{r3}` **or** `0`

I believe that now more directly matches the TRD text, and the answer is that `r2`'s value either meets that criteria or the syscall will return `INVALID`.